index.html

```diff
@@ -0,0 +1,475 @@
+
+## Data Model
+TK, draws heavily from federation, a little more general.
+
+### Realized types
+
+A type is *realized* wherever a resolver from a particular *subgraph* returns that type. We use the notation `subgraph::Type` to represent `Type` realized by `subgraph` and `subgraph::Type.field` to represent the resolver for `Type.field` within `subgraph`.
+
+```graphql
+#::[Query with resolvers and realized types]
+query {          # resolver                    | realizes
+                 # ----------------------------|--------------
+  rockets {      # rockets::Query.rockets     -> rockets::Rocket
+    captain {    # rockets::Rocket.astronaut  -> rockets::Astronaut
+      name       # astronauts::Astronaut.name -> String
+      rocket {   # rockets::Astronaut.rocket  -> rockets::Rocket
+        id       # rockets::Rocket.id         -> ID
+      }
+    }
+  }
+
+}
+```
+
+These distinctions are immaterial to the client. As far as clients are concerned, all selections on `Rocket` are the same.
+
+This illusion is maintained by the router, which must break incoming queries into subqueries. In the example above, the path `rockets.captain` is resolved by the `rockets::Rocket.astronaut` resolver, so its selection set is realized as a `rockets::Astronaut`. However, there is no `name` field for `rockets::Astronaut`—instead, the only resolver for `Astronaut.name` is `astronauts::Astronaut.name`. To process this query, the router must "convert" a `rockets::Astronaut` to an `astronauts::Astronaut` via an `_entities` query.
+
+### Portability
+A type is *portable to* a subgraph if it has a `@csdl_key` for that subgraph.
+
+### Free / Bound fields
+
+Free fields can be resolved by any subgraph, bound fields can only be resolved by the subgraph they're bound to.
+
+### Mandatory Types
+
+A CSDL document **must** define a `csdl_Graph` enum. Each enum value defines a subgraph. Each value **must** be annotated with a `@csdl_endpoint` directive specifying the endpoint URL for the subgraph.
+
+```graphql
+#::[Using `csdl_Graph` to define endpoints]
+enum csdl_Graph {
+  ROCKETS @csdl_endpoint(url: "https://rockets.example.com"),
+  ASTRONAUTS @csdl_endpoint(url: "https://astronauts.example.com"),
+}
+```
+
+The `csdl_Graph` enum is used as input to csdl directives which link fields and types to subgraphs.
+
+## Provided Types
+
+Valid CSDL documents **must** include the declarations of any CSDL types they use.
+
+### csdl_Never
+```graphql
+#::[Declaration of csdl_Never]
+type csdl_Never {}
+```
+
+The `csdl_Never` type is used to indicate the type of a field which **cannot be resolved**. This is typically due to a composition error. Producers **should** provide more specific error information with a `@csdl_Error` directive on the relevant field.
```

Comment:
Incorrect case on directive name.

Cool idea though. Is that new in this spec or have we been doing something like that before?